### DISCUSSION TITLE
Way to vertically align `import` and `from` in imports?

### DISCUSSION THREAD

**Discussion by dwhieb** [eslint-stylistic/eslint-stylistic#409] [Q&A]
Is there a rule that would allow me to require import statements to be vertically aligned on the `import` and `from` keywords?

I'd like for this:

```js
import { outputFile } from 'fs-extra'
import parseLinkHeader from 'parse-link-header'
import path from 'node:path'
import ProgressBar from 'progress'
```

... to be automatically formatted like this:

```js
import { outputFile }  from 'fs-extra'
import parseLinkHeader from 'parse-link-header'
import path            from 'node:path'
import ProgressBar     from 'progress'
```

If there's no way of doing this, I'll open an issue requesting it as a rule, and can probably put some time into coding it.

thanks!

**Comment by ljharb** [comment]
Conventions like this, with or without a rule, lead to massive diff churn as you add a bindings list that's longer than the others, or as you remove/adjust the longest bindings list so it's shorter. Alignment is rarely worth it.

**Reply by dwhieb** [reply]
Individual developers/teams decide whether that tradeoff is worth it to them for a particular style rule. Alignment is not objectively good/bad.

The purpose of Stylistic (it seems to me) is not to decide which rules are better/worse, but to enable developers to more easily work with the style rules of their choice—even when those choices are sometimes in the minority. (There are certainly plenty of other rules / rule options in Stylistic which probably get very little use, but are supported nonetheless.)

I doubt I'm the only person who finds the diff tradeoff worth it for vertical alignment, especially on smaller projects.

**Reply by ljharb** [reply]
Not all choices are equally valuable, though, and it's the job of eslint rule maintainers to exercise judgement sometimes and *not* provide all choices, for the benefit of all. There's a reason that alignment is very, very out of favor in the JS ecosystem, and at some point, the aggregate of subjectivity becomes sufficiently objective :-)

(to be clear, it's not my decision here, i'm just suggesting that the maintainers decline this one)
